### DISCUSSION TITLE
Adding current date to footer/header

### DISCUSSION THREAD

**Discussion by yosriayed** [marp-team/marp#553] [Q&A] [answered]
Is there a simple way to add today's date in the footer or header of my slides?

**Comment by yhatt** [comment]
Since Markdown is a static document format, you will need to use JavaScript, implement a Marp plugin, or apply preprocessing to the Markdown to achieve such functionality.

Check out similar question on StackOverflow: https://stackoverflow.com/questions/71458252/how-to-add-todays-date-in-marp-markdown

**Reply by yosriayed** [reply]
thanks. How to add the current date in the footer to all the slides? Something like how the pagination work

**Reply by yhatt** [reply, marked as answer]
This is a simple example with using JavaScript, based on the answe of StackOverflow. JavaScript replaces `<span class="date">` within footers in every page.

```markdown
---
marp: false
footer: '<span class="date"></span>'
---

Page 1

---

Page 2

---

Page 3

<script>
  window.addEventListener('DOMContentLoaded', function () {
    const now = new Date();
    for (const elm of document.querySelectorAll('footer .date')) {
      elm.textContent = now.toLocaleDateString();
    }
  })
</script>
```

**Reply by yosriayed** [reply]
Thank you!
